cylc/flow/scheduler.py

```diff
@@ -482,8 +482,25 @@ async def configure(self):
                 self.pool.set_stop_task(self.restored_stop_task_id)
         else:
             self.load_tasks_for_run()
-        if self.options.stopcp:
-            self.pool.set_stop_point(get_point(self.options.stopcp))
+
+        # Get stop after cycle point from config first, then options so that
+        # Options over-ride config.
+        if (
+            not self.is_restart or
+            not self.options.ignore_stopcp
+        ):
+            if (
+                'stop after cycle point' in self.config.cfg['scheduling'] and
+                self.options.stopcp is None
+            ):
+                self.pool.set_stop_point(get_point(
+                    self.config.cfg['scheduling']['stop after cycle point']
+                ))
+                self.options.stopcp = self.config.cfg['scheduling'][
+                    'stop after cycle point'
+                ]
+            if self.options.stopcp:
+                self.pool.set_stop_point(get_point(self.options.stopcp))
```

Comment:
Can this be simplified along the lines of:

```python
if not self.options.stopcp:
    # doesn't matter if stop after cycle point is None
    self.options.stopcp = self.config.cfg['scheduling']['stop after cycle point']
if self.options.stopcp:
    self.pool.set_stop_point(get_point(self.options.stopcp))
```

?

Reply:
Not really: Your suggestion would never allow the command line to over-ride the config: I'm not against tidying or commenting the logic though. It's a bit wierd.

Also, not having `'stop after cycle point' in self.config.cfg['scheduling']` would mean putting this in a `try:/except:` in case the config item was blank(I mean, I could set a default, but that seems weird in this instance).

Reply:
Sources of stopcp by priority
- Command line
- Database
- Config

We can simplify the logic by creating a temporary variable to store our stopcp in, and only overwriting the self.options.stopcp at the end of the process.

Reply:
> Not really: Your suggestion would never allow the command line to over-ride the config

I think it would, the config value is only read if the CLI opt is *not* set.